### ISSUE TITLE
Ubuntu 18.04 - master thread

### ISSUE THREAD

**Issue by ventz** [harvard-itsecurity/docker-misp#22]
Porting container to Ubuntu 18.04 and following official MISP project as closely as possible.

Outstanding issues:

* redis-server dies in 18.04 LTS

Problem:

It seems that using the specified config ("/usr/bin/redis-server /etc/redis/redis.conf") causes the server to start on localhost and attempt to bind to ipv6, which is not available, and thus the server errors with:

```
348:C 01 Jun 02:33:21.190 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
348:C 01 Jun 02:33:21.190 # Redis version=4.0.9, bits=64, commit=00000000, modified=0, pid=348, just started
348:C 01 Jun 02:33:21.190 # Configuration loaded
349:M 01 Jun 02:33:21.191 # Creating Server TCP listening socket ::1:6379: bind: Cannot assign requested address
```

@jbarlow-mcafee  - wanted to ping you since you had the config added in PR https://github.com/harvard-itsecurity/docker-misp/pull/14 -- was there a specific reason for adding the config file explicitly? It seems that by default the same set of config options get passed if you do not specify the config file, but then it will bind to localhost. Another alternative is to edit the /etc/redis/redis.conf on the fly to specify bind as "0.0.0.0" which also fixes this. If there's not a specific need for an option, I would rather not provide the config since the official version doesn't provide it. One potential reason for providing it and binding to "0.0.0.0" would be so that redis can be accessed directly. Thoughts on this?

**Comment by ventz** [comment]
I ended up going with the second (adding 0.0.0.0). Still curious if there was a specific need for the config just for my own information.

@jbarlow-mcafee - by the way, send me your email address - adding you to the README as a contributer.

**Comment by RANGERBEE** [comment]
can it be like Mysql - a setting in the env to allow traffic to localhost bind?
   turns out to be a docker compatibility thing with the new drivers..
  this may give a hint or two:  https://github.com/docker-library/mysql/issues/275#issuecomment-307782167

**Comment by ventz** [comment]
@RANGERBEE I think they may be running into a different issue from what I am understanding in that thread (a couple of different ones). I actually ran into something similar (in regards to the password) with mysql/mariadb, because it looks like on Ubuntu 18.04 LTS the default root password is blank by default, but that actually made it easier to make changes for MISP and then remove the root user.

For redis - If it supports an environmental variable of "bind-address" (or an equivalent), that would be much better. For now, I took this approach: 
https://github.com/harvard-itsecurity/docker-misp/blob/master/container/Dockerfile#L121

While still loading the config:
https://github.com/harvard-itsecurity/docker-misp/blob/master/container/supervisord.conf#L20-L24
(this way it allows someone to override the default config if they had to)

This way it allows people to override the redis config if they needed to, and it allows them also to expose the port directly to the outside if need be.

It's such a hard balance to make things easy/default while still allowing options/overrides along the way. Also - as I was mentioning in another thread to @jbarlow-mcafee - I am trying to stick to the original build process because semi-major upgrades become impossible to follow.

By the way - as of last night, the team at MISP gave us an official github repository to publish this container as their official docker one, so that should make things interesting :)

**Comment by RANGERBEE** [comment]
@ventz 
FYI  -  I did just try the phusion/baseimage:latest to address the zombie PID's and the multitude of reaping PID's thought..

All worked great with your latest Ubuntu:18.04 Dockerfile.

I did add one line to the file besides pulling the phusion image.
On line #26 I added 
```
RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

Simply housekeeping too keep image as tight as possible and things do build faster....


Thanks for all!
